### PR TITLE
Merge features/vs-for-mac-packages into master-vs-deps

### PR DESCRIPTION
**Pull request by jasonmalinowski**
This merges the entirety of features/vs-for-mac-packages (which includes the contents of #23731) into vs-for-mac-packages.

The individual changes in here were largely reviewed, this is just to have the PR ready when we've completed validation.

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
Tagging @Pilchie for ask mode approval. This is the VS for Mac payload we'll be merging in the next day or two to get it out of a private branch and back into our master branches.

We're still completing val build validation first.

**Comment by Pilchie** [comment]
Also tagging @MattGertz for this one due to impact, but I approve.

File: build/Targets/Versions.props [hunk 1]
```diff
@@ -14,7 +14,7 @@
     <!-- The release moniker for our packages.  Developers should use "dev" and official builds pick the branch
          moniker listed below -->
     <RoslynNuGetMoniker Condition="'$(RoslynNuGetMoniker)' == ''">dev</RoslynNuGetMoniker>
-    <RoslynNuGetMoniker Condition="'$(OfficialBuild)' == 'true'">beta3</RoslynNuGetMoniker> 
+    <RoslynNuGetMoniker Condition="'$(OfficialBuild)' == 'true'">vs-for-mac</RoslynNuGetMoniker>
```
Comment (jasonmalinowski) [inline build/Targets/Versions.props]: We won't merge this one line, obviously. :smile:

**Comment by jasonmalinowski** [comment]
@dotnet-bot retest windows_debug_unit64_prtest please.

**Pull request merged**